catroid/src/main/java/org/catrobat/catroid/ui/BaseActivity.java

```diff
@@ -123,6 +128,22 @@ public boolean onOptionsItemSelected(MenuItem item) {
 		return super.onOptionsItemSelected(item);
 	}
 
+	public boolean isRecoveredFromCrash() {
+		return PreferenceManager.getDefaultSharedPreferences(this).getBoolean(RECOVERED_FROM_CRASH, false);
+	}
```

Comment:
Please make private and put it after the `checkIfCrashRecoveryAndFinishActivity` method.

> We like to put the private utilities called by a public function right after the public function itself. This follows the stepdown rule and helps the program read like a newspaper article. -Clean Code|Chapter 10: Classes